base_layer/wallet_ffi/src/lib.rs

```diff
@@ -1833,22 +1872,21 @@ pub unsafe extern "C" fn wallet_test_complete_sent_transaction(
     }
 }
 
-/// This function will simulate the process when a completed transaction is detected as mined on
-/// the base layer. The function will update the status of the completed transaction AND complete
-/// the transaction on the Output Manager Service which will update the status of the outputs
+/// This function will simulate the process when a completed transaction is broadcast to
+/// the base layer mempool. The function will update the status of the completed transaction
 ///
 /// ## Arguments
 /// `wallet` - The TariWallet pointer
-/// `tx` - The TariCompletedTransaction pointer
+/// `tx_id` - The TxId of the transaction to operate on
 /// `error_out` - Pointer to an int which will be modified to an error code should one occur, may not be null. Functions
 /// as an out parameter.
 ///
 /// ## Returns
 /// `bool` - Returns if successful or not
 #[no_mangle]
-pub unsafe extern "C" fn wallet_test_mined(
+pub unsafe extern "C" fn wallet_test_finalize_received_transaction(
     wallet: *mut TariWallet,
-    tx: *mut TariCompletedTransaction,
+    tx_id: c_ulonglong,
```

Comment:
This is probably best left as ```tx: *mut TariCompletedTransaction``` and then extracting the tx_id from it for the following reasons:
1) We can enforce what type of object enters the FFI vs trying to check if the ID is valid and is of a completed transaction.
2) The client app would have had to have constructed the object previously in order to retrieve the tx_id from it.
3) Only a null check is needed against the pointer.

Reply:
agreed, done